capabilities/src/execute_task_solution_capability.h

```diff
@@ -58,13 +58,23 @@ class ExecuteTaskSolutionCapability : public MoveGroupCapability
 	void initialize() override;
 
 private:
-	bool constructMotionPlan(const moveit_task_constructor_msgs::Solution& solution,
+	using ActionType = moveit_task_constructor_msgs::action::ExecuteTaskSolution;
```

Comment:
I agree that this makes things much more readable. Maybe just use `ExecuteTaskSolution` directly?